src/language/providers/utils.ts

```diff
@@ -283,6 +291,33 @@ export function checkSetVariableOpen(
   return false
 }
 
+export function getNsPrefix(
+  document: vscode.TextDocument,
+  position: vscode.Position
+) {
+  var initialLineNum = position.line
+  var lineNum = 0
+  while (initialLineNum !== 0 && lineNum <= initialLineNum) {
+    const lineText = document
+      .lineAt(lineNum)
+      .text.substr(0, document.lineAt(lineNum).range.end.character)
+    // returns an empty prefix value
+    if (lineText.includes('<schema')) {
+      return ''
+    }
+    // returns the prefix attached to the schema element plus a colon
+    if (lineText.match(new RegExp('<.*:schema'))) {
+      return lineText.substring(
+        lineText.indexOf('<') + 1,
+        lineText.indexOf(':') + 1
+      )
+    }
```

Comment:
In the other review I suggested a regex that matches both `<foo:schema` and `<schema` so you don't need to do two scans, as well as using a capture group so you don't need to use substring. Did that not work?

Also, maybe move this RegExp to some global state so it doesn't need to be recompiled for iteration everytime this function is called? Feels like taht could be cause a performance hit?